File: app-presentation/src/main/java/com/zxjdev/smile/presentation/user/settings/SettingsContract.java
package com.zxjdev.smile.presentation.user.settings;

import com.zxjdev.smile.presentation.common.base.BasePresenter;
import com.zxjdev.smile.presentation.common.base.BaseView;

public interface SettingsContract {

  interface View extends BaseView {

    void onLogoutSuccess();
  }

  interface Presenter extends BasePresenter<View> {

    void handleLogout();
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/user/settings/SettingsContract.kt
package com.zxjdev.smile.presentation.user.settings

import com.zxjdev.smile.presentation.common.base.BasePresenter
import com.zxjdev.smile.presentation.common.base.BaseView

interface SettingsContract {

    interface View : BaseView {

        fun onLogoutSuccess()
    }

    interface Presenter : BasePresenter<View> {

        fun handleLogout()
    }
}
